tools/funccount.py

```diff
@@ -57,18 +57,25 @@ def signal_ignore(signal, frame):
 bpf_text = """
 #include <uapi/linux/ptrace.h>
 
-struct key_t {
+#define MAP_SIZE 10000
+
+struct val_t {
     u64 ip;
+    u64 value;
 };
-BPF_HASH(counts, struct key_t);
+BPF_TABLE("array", int, struct val_t, counts, MAP_SIZE);
 
 int trace_count(struct pt_regs *ctx) {
     FILTER
-    struct key_t key = {};
-    u64 zero = 0, *val;
-    key.ip = ctx->ip;
-    val = counts.lookup_or_init(&key, &zero);
-    (*val)++;
+    struct val_t *val;
+    u64 zero = 0;
+    u32 key = ctx->ip % MAP_SIZE;
```

Comment:
What I'm worried about is that in case of collision, the user really doesn't have any choice but to trace fewer and fewer functions. And if two functions out of 50 end up colliding, it's not going to be very pleasant figuring out which.